geoips/commandline/commandline_interface.py

```diff
@@ -83,9 +84,50 @@ def execute_command(self):
             sys.exit(2)
 
 
+def support_legacy_procflows():
+    """Run a series of checks on sys.argv to support legacy calls to run_procflow.
+
+    The new GeoIPS CLI has changed how we run our process workflows from now on. Instead
+    of calling 'run_procflow' or 'data_fusion_procflow', we now call
+    'geoips run single_source' or 'geoips run data_fusion'. This function parses through
+    sys.argv and performs the necessary translations to match our current format so that
+    the CLI's argparser can call the appropriate functionality.
+
+    Returns
+    -------
+    legacy: bool
+        - The truth value as to whether or not a legacy procflow call was used
+    """
+    defined_procflow = None
+    if "run_procflow" == basename(sys.argv[0]):
+        entrypoint = "run_procflow"
+        defined_procflow = "single_source"
+
+    elif "data_fusion_procflow" == basename(sys.argv[0]):
+        entrypoint = "data_fusion_procflow"
+        defined_procflow = "data_fusion"
+
+    if defined_procflow:
+        # Either 'run_procflow' or 'data_fusion_procflow' was called, make the
+        # appropriate translations in sys.argv so the CLI's parser can call the correct
+        # functionality
+        if "--procflow" in sys.argv:
+            # If '--procflow' was found in the command line arguments, loop through the
+            # arguments and grab the correct procflow.
+            for idx, arg in enumerate(sys.argv):
+                if arg == "--procflow" and len(sys.argv) > idx + 1:
+                    defined_procflow = sys.argv[idx + 1]
+        sys.argv[0] = sys.argv[0].replace(entrypoint, "geoips")
+        sys.argv.insert(1, "run")
+        sys.argv.insert(2, defined_procflow)
+        return True
+    return False
+
+
 def main():
     """Entry point for GeoIPS command line interface (CLI)."""
-    geoips_cli = GeoipsCLI()
+    legacy = support_legacy_procflows()
+    geoips_cli = GeoipsCLI(legacy)
```

Comment:
Wouldn't this need to be `GeoipsCLI(legacy=legacy)`? This looks like it would go to `instructions_dir`.

Maybe we need to add one integration test to ensure that legacy calls work correctly. Or a unit test to ensure that GeoipsCLI receives its arguments correctly.

Reply:
Yup. Let me do this real quick. See #598 for a discussion about legacy procflows.